crates/polars-plan/src/logical_plan/mod.rs

```diff
@@ -256,6 +256,12 @@ pub enum LogicalPlan {
     },
 }
 
+impl PartialEq for LogicalPlan {
```

Comment:
I don't think we should do this. Now we lose trait protection. Why is this needed? 

I think we shoud use a newtype for this.

Reply:
`Expr` implements PartialEq and now an `Expr` will have a `LogicalPlan` ie: `Expr::SubPlan(Box<LogicalPlan>...)`
What would a newtype look like? Not sure what you mean by that.

https://doc.rust-lang.org/rust-by-example/generics/new_types.html?

Reply:
> I don't think we should do this. Now we lose trait protection. Why is this needed?
> 
> I think we shoud use a newtype for this.

Made a change, lmk if this is what you're looking for.